common/tracing.py

```diff
@@ -118,3 +120,24 @@ def set_span_attributes(span, attributes):
         clean_attrs[k] = v
 
     span.set_attributes(clean_attrs)
+
+
+@contextmanager
+def time_for_span(attribute_name: str, span=None):
```

Comment:
nitpick: not sure about the name. 

How about `duration_as_span_attr`? Bit mechanical, but maybe easier to understand?